server/models/NodeModel.ts

```diff
@@ -1,10 +1,10 @@
-import { Schema, model, Types } from "mongoose";
-import { TreeNode } from "../types/datamodels";
+import { Schema, model, Types } from 'mongoose';
```

Comment:
this should have been updated. Can you ensure it's not in order to minimize touching unrelated files?

Reply:
This branch needs merging with the work that has been pushed to dev, I am updating it now.